core/rawdb/accessors_indexes.go

```diff
@@ -180,6 +207,62 @@ func ReadReceipt(db ethdb.Reader, hash common.Hash, config *params.ChainConfig)
 	return nil, common.Hash{}, 0, 0
 }
 
+// ReadRawReceipt allows reading a raw receipt and a given position. Returns the necessary data to derive a complete
+// receipt as well.
+func ReadRawReceipt(db ethdb.Reader, blockHash common.Hash, blockNumber, blockIndex uint64) (*types.Receipt, uint64, uint) {
```

Comment:
The extra return values for gas and log count should be in a struct. You can return this struct by-value. The idea here is that we might have to extract more values later and we don't want to modify the return values.